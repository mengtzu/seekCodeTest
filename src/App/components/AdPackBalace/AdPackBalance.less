@import (reference) "~seek-style-guide/theme";

.root {
    margin-top: @grid-row-height * 2;
}

.packBalanceWrapper {
    display: flex;
}

.packBalanceText {
    width: @grid-column-width * 2;
}

.packBalanceQuantity {
    flex-grow: 1;
}