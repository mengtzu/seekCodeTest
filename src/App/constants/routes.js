const root = '/';
const adPacks = '/adPacks';
const checkout = '/checkout';
const confirmation = '/confirmation';
const logout = '/Login/Logout';

export const routes = {
    root,
    adPacks,
    checkout,
    confirmation,
    logout
};

export default routes;
