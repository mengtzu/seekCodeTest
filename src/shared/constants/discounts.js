export const DISCOUNT_TYPE_FLAT_RATE = 'flatRate';
export const DISCOUNT_TYPE_THRESHOLD = 'threshold';
export const DISCOUNT_TYPE_X_FOR_Y = 'xfory';
