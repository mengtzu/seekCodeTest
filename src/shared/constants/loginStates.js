export const AUTH_STATUS_UNAUTHENTICATED = 'pending'; //'unauthenticated' exposes the real SEEK login, don't want that!
export const AUTH_STATUS_AUTHENTICATED = 'authenticated';
