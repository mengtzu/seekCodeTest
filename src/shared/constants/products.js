export const CLASSIC = 'classic';
export const STANDOUT = 'standout';
export const PREMIUM = 'premium';
