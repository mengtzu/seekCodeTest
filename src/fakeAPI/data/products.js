import { CLASSIC, STANDOUT, PREMIUM } from '../../shared/constants/products';

let productsData = {};

//Bracket notation isn't necessarily preferred but here it lets us avoid having to iterate over this object in various places

productsData[CLASSIC] = {
    id: CLASSIC,
    basePrice: 269.99
};

productsData[STANDOUT] = {
    id: STANDOUT,
    basePrice: 322.99
};

productsData[PREMIUM] = {
    id: PREMIUM,
    basePrice: 394.99
}

export default productsData;