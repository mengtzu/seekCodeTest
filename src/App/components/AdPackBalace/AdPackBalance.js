import React from 'react';
import { connect } from 'react-redux';
import {
    PageBlock,
    Card,
    Section,
    Text
} from 'seek-style-guide/react';
import styles from './AdPackBalance.less';

import { CLASSIC, STANDOUT, PREMIUM } from '../../../shared/constants/products';

const mapStateToProps = (state) => {
    return {
        adStock: state.advertiser.adStock
    };
};

const renderPackBalance = (label, quantity) => {
    return (
        <div className={styles.packBalanceWrapper}>
            <Text strong className={styles.packBalanceText}>{label}:</Text>
            <Text className={styles.packBalanceQuantity}>{quantity}</Text>
        </div>
    )
};

export const AdPackBalance = ({ adStock }) => {
    return (
        <div className={styles.root}>
            <PageBlock>
                <Card>
                    <Section>
                        <Text heading>Your available ads</Text>
                        {renderPackBalance('Classic', adStock[CLASSIC])}
                        {renderPackBalance('Standout', adStock[STANDOUT])}
                        {renderPackBalance('Premium', adStock[PREMIUM])}
                    </Section>
                </Card>
            </PageBlock>
        </div>
    )
};

export default connect(mapStateToProps)(AdPackBalance);