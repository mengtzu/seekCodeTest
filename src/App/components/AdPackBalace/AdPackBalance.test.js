import React from 'react';
import { AdPackBalance } from './AdPackBalance';
import renderer from 'react-test-renderer';
import { CLASSIC, STANDOUT, PREMIUM } from '../../../shared/constants/products';

const adStock = {};
adStock[CLASSIC] = 32;
adStock[STANDOUT] = 0;
adStock[PREMIUM] = 777;

it('renders correctly', () => {
    const tree = renderer.create(
        <AdPackBalance adStock={adStock}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
