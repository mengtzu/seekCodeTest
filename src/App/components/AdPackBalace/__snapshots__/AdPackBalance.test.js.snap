// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="root"
>
  <div
    className=""
  >
    <div
      className="content"
    >
      <div
        className="root"
      >
        <div
          className="root"
        >
          <div
            className="root heading"
          >
            <span
              className=""
            >
              Your available ads
            </span>
          </div>
          <div
            className="packBalanceWrapper"
          >
            <div
              className="root packBalanceText"
            >
              <span
                className="root"
              >
                Classic
                :
              </span>
            </div>
            <div
              className="root packBalanceQuantity"
            >
              <span
                className=""
              >
                32
              </span>
            </div>
          </div>
          <div
            className="packBalanceWrapper"
          >
            <div
              className="root packBalanceText"
            >
              <span
                className="root"
              >
                Standout
                :
              </span>
            </div>
            <div
              className="root packBalanceQuantity"
            >
              <span
                className=""
              >
                0
              </span>
            </div>
          </div>
          <div
            className="packBalanceWrapper"
          >
            <div
              className="root packBalanceText"
            >
              <span
                className="root"
              >
                Premium
                :
              </span>
            </div>
            <div
              className="root packBalanceQuantity"
            >
              <span
                className=""
              >
                777
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
